Sources/NIO/System.swift

```diff
@@ -150,18 +150,26 @@ private func preconditionIsNotUnacceptableErrno(err: CInt, where function: Strin
 @inline(__always)
 @discardableResult
 internal func syscall<T: FixedWidthInteger>(blocking: Bool,
+                                            eprototypeWorkaround: Bool = false,
                                             where function: String = #function,
                                             _ body: () throws -> T)
         throws -> IOResult<T> {
     while true {
         let res = try body()
         if res == -1 {
             let err = errno
-            switch (err, blocking) {
-            case (EINTR, _):
+            switch (err, blocking, eprototypeWorkaround) {
+            case (EINTR, _, _):
                 continue
-            case (EWOULDBLOCK, true):
+            case (EWOULDBLOCK, true, _):
                 return .wouldBlock(0)
+            #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+            case (EPROTOTYPE, _, eprototypeWorkaround):
```

Comment:
Is this meant to be `true` rather than `eprototypeWorkaround`?

Reply:
Uhhhhh yes.